src/OmniSharp.LanguageServerProtocol/Helpers.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using OmniSharp.Extensions.LanguageServer.Models;
+using OmniSharp.Models;
+using OmniSharp.Models.Diagnostics;
+
+namespace OmniSharp.LanguageServerProtocol
+{
+    public static class Helpers
+    {
+        public static Diagnostic ToDiagnostic(this DiagnosticLocation location)
+        {
+            return new Diagnostic()
+            {
+                // We don't have a code at the moment
+                // Code = quickFix.,
+                Message = location.Text,
+                Range = location.ToRange(),
+                Severity = ToDiagnosticSeverity(location.LogLevel),
+                // TODO: We need to forward this type though
+                Source = "csharp",
+            };
+        }
+
+        public static Range ToRange(this QuickFix location)
+        {
+            return new Range()
+            {
+                Start = new Position()
+                {
+                    Character = location.Column,
+                    Line = location.Line
+                },
+                End = new Position()
+                {
+                    Character = location.EndColumn,
+                    Line = location.EndLine
+                },
+            };
+        }
+
+        public static DiagnosticSeverity ToDiagnosticSeverity(string logLevel)
+        {
+            // We stringify this value and pass to clients
+            // should probably use the enum at somepoint
+            if (Enum.TryParse<Microsoft.CodeAnalysis.DiagnosticSeverity>(logLevel, out var severity))
+            {
+                switch (severity)
+                {
+                    case Microsoft.CodeAnalysis.DiagnosticSeverity.Error:
+                        return DiagnosticSeverity.Error;
+                    case Microsoft.CodeAnalysis.DiagnosticSeverity.Hidden:
+                        return DiagnosticSeverity.Hint;
```

Comment:
FWIW, these aren't the same. In VS Code, `Hint` still causes diagnostics to appear in the error list, which is not what `Hidden` is for.

Reply:
Should we do anything with hidden ones at the moment?

https://github.com/Microsoft/language-server-protocol/issues/325

Reply:
Here's what we do in VS Code: https://github.com/OmniSharp/omnisharp-vscode/blob/master/src/features/diagnosticsProvider.ts#L294-L304.

Essentially, we elevate hidden diagnostics to "Info" and then have an option to allow users to disable them.